src/controllers/bucket-controller.js

```diff
@@ -0,0 +1,68 @@
+const { BucketModel, ProductModel } = require('../models');
+const { NotFoundError, ValidationError } = require('../errors');
+
+const { ResponseHandlerUtil } = require('../utils');
+
+async function addProductToBucket(req, res, next) {
+  const { productId, quantity } = req.body;
+  const userId = req.userData._id;
+
+  try {
+    const product = await ProductModel.findOne({ _id: productId });
+    const increaseAmount = product.price * quantity;
+
+    let bucketData;
+
+    if (!product) {
+      throw new NotFoundError('Product not found');
+    }
+    if (product.quantity < quantity) {
+      throw new ValidationError('Product not available in that quantity');
+    }
+
+    const currentBucketData = await BucketModel.findOne({ userId });
+    const pickedProduct = currentBucketData.products.find(
+      (prod) => prod.productId.toString() === productId.toString()
```

Comment:
productId.toString()

productId is already string, do it only for doc id field